src/keydra/providers/gitlab.py

```diff
@@ -0,0 +1,175 @@
+import copy
+import json
+
+import yaml
+from schema import And, Optional, Or, Regex, Schema, SchemaError, Use
+
+from keydra.clients.gitlab import GitlabClient
+from keydra.exceptions import (ConfigException, DistributionException,
+                               RotationException)
+from keydra.logging import get_logger
+from keydra.providers.base import (BaseProvider, ConfigProvider,
+                                   exponential_backoff_retry)
+
+LOGGER = get_logger()
+SPEC_SCHEMA = Schema({'provider': 'gitlab',
+                      'key': Regex(r'^[A-Za-z0-9_{}]{1,255}$'),
+                      'source': And(str, Use(str.lower), lambda s: s in ('key', 'secret')),
+                      'config': {
+                          'repository': Or([And(str, len)], And(str, len)),
+                          'scope': 'repository',
+                          Optional('environment'): And(str, len)},
+                      'envs': [And(str, len)]
+                      })
+
+
+class Client(BaseProvider):
+
+    def __init__(self, session=None, credentials=None, **kwargs):
+        self.repo_client = GitlabClient(
+            access_token=credentials["access_token"]
+        )
+
+    def _distribute_repository_secret(self, secret, dest):
+        config = dest['config']
+
+        try:
+            self.repo_client.set_repo_variable(
+                repo_name=config['repository'],
+                key=dest['key'],
+                value=secret[dest['source']]
+            )
+
+        except Exception as e:  # pragma: no cover
+            LOGGER.warn('Failed to distribute secret to Gitlab repo {}: {}'.format(
+                config['repository'], e))
+            raise DistributionException(e)
+
+        LOGGER.info(
+            'Successfully distributed {}.{}.key to '
```

Comment:
This may be a little misleading -- the distributed value is not always `key`, but rather `dest['source']`, right?

Reply:
I kept the log message identical with its counterpart from the github provider (https://github.com/Athena-Home-Loans/keydra/blob/9bab1b21e025ceb6ae074ea936d693e36efae5a4/src/keydra/providers/github.py#L58)

Happy change of course, do you have a suggestion?

Reply:
`'Successfully distributed {}.{}.{} to '` with `dest['source']` added as a parameter to `.format()` should cut it 🙂